File: Sorting Algorithms/Selection Sort/SelectionSort.c
#include <stdio.h>


int* selectionSort(int*,int);


int main(void){
    int n = 10;
    int arr[] = {2, 1, 5, 8, 3, 9, 4, 7, 6, 0};
    printf("Selection Sort Algorithm using C");

    printf("\n\nArray before sorting: [ ");
    for(int i=0; i<n-1; i++){
        printf("%d, ", arr[i]);
    }
    printf("%d ]", arr[n-1]);

    selectionSort(arr, n);

    printf("\n\nArray after sorting: [ ");
    for(int i=0; i<n-1; i++){
        printf("%d, ", arr[i]);
    }
    printf("%d ]", arr[n-1]);
    
    return 0;
}


int* selectionSort(int* arr, int n){
    int temp, k;
    for(int i=0; i<n-1; i++){
        k = i;
        for(int j=i+1; j<n; j++){
            if(arr[j] < arr[k]){
                k = j;
            }
        }
        temp = arr[i];
        arr[i] = arr[k];
        arr[k] = temp;      
    }
    return arr;
}

File: Sorting Algorithms/Selection Sort/SelectionSort.java
class SelectionSort{


    public static void main(String[] args){
        int arr[] = {2, 1, 5, 8, 3, 9, 4, 7, 6, 0};
        int n = arr.length;

        System.out.println("Selection Sort Algorithm using Java");

        System.out.print("\n\nArray before sorting: [ ");
        for(int i=0; i<n-1; i++){
            System.out.print(arr[i] + ", ");
        }
        System.out.println(arr[n-1]+ " ]");

        selectionSort(arr, n);

        System.out.print("\n\nArray after sorting: [ ");
        for(int i=0; i<n-1; i++){
            System.out.print(arr[i] + ", ");
        }
        System.out.println(arr[n-1]+ " ]");
    }


    public static void selectionSort(int[] arr, int n){
        int k, temp;
        for(int i=0; i<n-1; i++){
            k = i;
            for(int j=i+1; j<n; j++){
                if(arr[j] < arr[k]){
                    k = j;
                }
            }
            temp = arr[i];
            arr[i] = arr[k];
            arr[k] = temp;
        }
    }
    

}